### PR TITLE
GEODE-6624: fix nested exception causing SIGABRT

### PR DESCRIPTION
**Pull request by pdxcodemonkey**

### CODE REVIEW

File: cppcache/CMakeLists.txt [hunk 1]
```diff
@@ -59,6 +59,8 @@ elseif ("Windows" STREQUAL ${CMAKE_SYSTEM_NAME} )
   set( _WIN32 1 )
 endif()
 
+get_target_property(JAVAOBJECT_JAR_PATH javaobject JAR_FILE)
```
Comment (pivotal-jbarrett) [inline cppcache/CMakeLists.txt]: This project doesn't depend on javaobject so it is relying on ordering of targets to complete correctly. Make this target depend on the Javadoc target. Also, does the root level project for C++ really need this? I imagine it is really just for integration tests. It should really be moved there.

File: cppcache/integration/framework/config.h.in [hunk 2]
```diff
@@ -21,5 +21,6 @@
 #define INTEGRATION_TEST_FRAMEWORK_CONFIG_H
 
 #define GFSH_EXECUTABLE "@Geode_gfsh_EXECUTABLE@"
+#define JAVAOBJECT_JAR_PATH "@JAVAOBJECT_JAR_PATH@"
```
Comment (pivotal-jbarrett) [inline cppcache/integration/framework/config.h.in]: This is the project where the javaobject project should be dependent and variable defined.

File: cppcache/integration/test/CMakeLists.txt [hunk 3]
```diff
@@ -38,6 +38,7 @@ target_compile_definitions(cpp-integration-test
 target_include_directories(cpp-integration-test
   PUBLIC
    ${CMAKE_CURRENT_BINARY_DIR}
+   ${CMAKE_CURRENT_BINARY_DIR}/..
```
Comment (pivotal-jbarrett) [inline cppcache/integration/test/CMakeLists.txt]: Wait, what? Why is this changing?

File: cppcache/integration/test/FunctionExecutionTest.cpp [hunk 4]
```diff
@@ -125,3 +128,33 @@ TEST(FunctionExecutionTest, UnknownFunctionAsyncOnRegion) {
                    .execute("I_Don_t_Exist"),
                FunctionExecutionException);
 }
+
+TEST(DISABLED_FunctionExecutionTest,
```
Comment (pivotal-jbarrett) [inline cppcache/integration/test/FunctionExecutionTest.cpp]: Why are we adding a disabled test?

Comment (pdxcodemonkey) [reply cppcache/integration/test/FunctionExecutionTest.cpp]: This test reproduces the issue, which includes throwing the _wrong_ exception, aka NotConnectedException.  If I enable the test, I have to change the expected exception to NotConnectedException and change the repro instructions for the server bug (GEODE-6699) to include "change the expected exception to <blah> and rebuild NC tests".  Our thinking was to just leave it like this for the Geode team, I take it you think it should be the other way 'round?

Comment (pivotal-jbarrett) [reply cppcache/integration/test/FunctionExecutionTest.cpp]: I would suggest writing a JIRA to address this once GEODE-6699 is complete.

Comment (pivotal-jbarrett) [reply cppcache/integration/test/FunctionExecutionTest.cpp]: I would also leave a not saying why this is disabled. I would also create a version of the test that asserts the current "correct" behavior that can be removed when the server is corrected.

File: cppcache/integration/test/CMakeLists.txt (hunk 3 above)
Comment (pdxcodemonkey) [reply cppcache/integration/test/CMakeLists.txt]: config.h is placed in ${CMAKE_CURRENT_BINARY_DIR}/../framework by the cmake configure step.

Comment (pivotal-jbarrett) [reply cppcache/integration/test/CMakeLists.txt]: Right, but you shouldn't be using `config.h` directly outside of the framework. Is there not a framework class to access these values? Similar to how we do in .NET.

File: cppcache/integration/test/FunctionExecutionTest.cpp [hunk 5]
```diff
@@ -26,10 +27,12 @@
 
 #include "framework/Cluster.h"
 #include "framework/Gfsh.h"
+#include "framework/config.h"
```
Comment (pivotal-jbarrett) [inline cppcache/integration/test/FunctionExecutionTest.cpp]: Maybe don't access the config.h macros directly. They configure the framework. Add a class to the framework that provides and interface into specific framework values.

File: cppcache/integration/test/FunctionExecutionTest.cpp (hunk 4 above)
Comment (pdxcodemonkey) [reply cppcache/integration/test/FunctionExecutionTest.cpp]: Ima just enable the test and assert the "incorrect" exception, and add a note to the server bug that they'll have to change this line of our test code and rebuild it to repro.  Much less fuss that way.

**Pull request merged**